crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -1253,14 +1253,19 @@ impl<T: PoolTransaction> AllTransactions<T> {
         &self,
         transaction: ValidPoolTransaction<T>,
     ) -> Result<ValidPoolTransaction<T>, InsertErr<T>> {
-        if !transaction.origin.is_local() || self.local_transactions_config.no_local_exemptions() {
+        if !transaction.origin.is_local() ||
+            self.local_transactions_config.no_local_exemptions() ||
+            self.local_transactions_config
+                .contains_local_address(transaction.transaction.sender())
```

Comment:
I want to move these checks to a function in LocalTransactionConfig that accepts an Origin and then returns true if it is considered local

Reply:
Ok, but still figuring the failing test